### PR TITLE
VACMS-10586: Improve pre-commit branch name check.

### LINKED ISSUE #10586: Review pre-commit check of branch naming convention.
## Description
This is a frequent annoyance with Dependabot PRs that require local modification:

Screen Shot 2022-09-01 at 10 46 29 AM

The normal action is to then run `git commit --no-verify`, an obvious antipattern.

If this was ever useful, it's less useful now, because:

- Dependabot does not follow this convention.
- Reversions performed in the GitHub UI do not follow this convention.
- Codespaces does not follow this convention.
- This isn't enforced on GitHub itself, so you can just do `--no-verify` to bypass it if you have a rawr yum branch name you'd like to use

In my opinion it just adds additional friction for adding local commits on Dependabot PRs and possibly for dealing with emergency situations like reverts of earlier PRs, and again the only practical option for bypassing it is also bypassing the pre-commit hook in its entirety.

## Acceptance Criteria
- [ ] I've begged the team as a whole to abandon this process, and they've either ignored me, berated me, or blandly agreed.

### PR DESCRIPTION
**Pull request by ndouglas**
## Description

Closes #10586.

**TL;DR**: This is so we don't have to type `--no-verify` to commit when we're working on a Dependabot or revert PR locally.

I think there are only a few branch name patterns we wanna support, so this is probably a reasonable approach.

I'll write a test script to show what branch names are approved and which are rejected.

### Test Script
```bash
#!/bin/bash

function check_branch_name() {
  case $1 in 
    VAGOV-TEAM-[0-9]* | VACMS-[0-9]* | dependabot/* | revert-*) return 0;;
    *) return 1;;
  esac
}

branch_names=(
  VACMS-0000-Use-ProphecyTrait
  VACMS-10586-Improve-precommit-branch-name-check
  VACMS-10870-upgrade-to-php81_rebased
  VACMS-11049-Fix-attributes-error
  VACMS-11082-Investigate-gathering-line-count-metrics
  VACMS-11497-Set-last-touched-by-a-human-field-on-all-published-nodes
  VACMS-11639-Fix-status-error-processing
  VACMS-11669-Fix-warning-importing-configuration
  VACMS-11761-Delay-accessibility-tests-to-allow-CSS-transitions-to-finish
  VACMS-11811-Shifts-PHP-Lint-into-GHA-action
  VACMS-11881-Change-order-of-file-permission-changes
  VACMS-11896-Make-contentreleasestatuscontroller-test-more-reliable
  VACMS-11904-Undefined-array-key-href
  VACMS-11905-Undefined-key-req_optional
  VACMS-11906-Patch-taxonomy_entity_index-to-fix-error
  VACMS-8671-Part-2
  dependabot/composer/drupal/dynamic_entity_reference-3.0.0-beta1
  dependabot/composer/aws/aws-sdk-php-3.250.0
  dependabot/composer/aws/aws-sdk-php-3.252.5
  dependabot/composer/aws/aws-sdk-php-3.253.1
  dependabot/composer/aws/aws-sdk-php-3.254.0
  dependabot/composer/consolidation/site-process-4.2.1
  dependabot/composer/cweagans/composer-patches-1.7.2
  dependabot/composer/datadog/dd-trace-0.79.0
  dependabot/composer/datadog/dd-trace-0.80.0
  dependabot/composer/drupal/allowed_formats-1.5.0
  dependabot/composer/drupal/animated_gif-1.7.0
  dependabot/composer/drupal/auto_entitylabel-3.0.0-beta4
  dependabot/composer/drupal/config_split-2.0.0-rc4
  dependabot/composer/drupal/core-recommended-9.3.11
  dependabot/composer/drupal/core-recommended-9.3.12
  dependabot/composer/drupal/core-recommended-9.3.13
  dependabot/composer/drupal/core-recommended-9.3.14
  dependabot/composer/drupal/core-recommended-9.3.15
  dependabot/composer/drupal/core-recommended-9.3.9
  dependabot/composer/drupal/core-recommended-9.4.1
  dependabot/composer/drupal/core-recommended-9.4.3
  dependabot/composer/drupal/ctools_block-4.0.1
  dependabot/composer/drupal/danse-2.2.7
  dependabot/composer/drupal/danse_content_moderation-2.0.0-beta3
  dependabot/composer/drupal/devel-4.1.5
  dependabot/composer/drupal/dropzonejs-2.7.0
  dependabot/composer/drupal/entity_embed-1.1.0
  dependabot/composer/drupal/fast_404-2.0.0-alpha5
  dependabot/composer/drupal/geocoder-3.31.0
  dependabot/composer/drupal/hook_event_dispatcher-3.3.1
  dependabot/composer/drupal/menu_breadcrumb-1.13.0
  dependabot/composer/drupal/metatag-1.22.0
  dependabot/composer/drupal/migrate_source_csv-3.5.0
  dependabot/composer/drupal/next-1.3.1
  dependabot/composer/drupal/next-1.4.0
  dependabot/composer/drupal/next-1.6.0
  dependabot/composer/drupal/node_link_report-1.18.0
  dependabot/composer/drupal/password_policy-3.1.0
  dependabot/composer/drupal/path_redirect_import-2.0.5
  dependabot/composer/drupal/pathauto-1.10.0
  dependabot/composer/drupal/pathauto-1.11.0
  dependabot/composer/drupal/pathauto-1.9.0
  dependabot/composer/drupal/redirect-1.6.0
  dependabot/composer/drupal/role_delegation-1.1.0
  el-delete-composer-cache-dir
  el-php-81-disable-memcache-brd-docs
  integration-VACMS-8877-homepage-hardening-refresh
  integration-covid-status
  integration-lovell
  integration-recurring-events
  integration-tour-ui
  joshworkscivicactions-patch-1
  main
  master
  mchelen-gov-patch-1
  neil-demo
  next-drupal-poc
  nh_composer_package
  revert-10362-dependabot/composer/drupal/smart_date-3.6.0
  revert-11949-VACMS-11948-gha_command_deprecation
  revert-9089-VACMS-8649-next_news_story
  revert-9369-revert-9345-VACMS-9097-new-facility-email
  revert-9459-revert-9401-VACMS-8588-remove-lando
  VACMS-invalid
  VAGOV_TEAM-2304204923-invalid
)

for branch_name in "${branch_names[@]}"; do
  status="$(check_branch_name "${branch_name}" && echo "PASS" || echo "FAIL")";
  printf "%80s %s\n" ${branch_name}      ${status}
done;
```

### Test Script Output

```
$ ./test_script.sh 
                                                    VACMS-0000-Use-ProphecyTrait PASS
                                 VACMS-10586-Improve-precommit-branch-name-check PASS
                                            VACMS-10870-upgrade-to-php81_rebased PASS
                                                VACMS-11049-Fix-attributes-error PASS
                            VACMS-11082-Investigate-gathering-line-count-metrics PASS
            VACMS-11497-Set-last-touched-by-a-human-field-on-all-published-nodes PASS
                                         VACMS-11639-Fix-status-error-processing PASS
                                 VACMS-11669-Fix-warning-importing-configuration PASS
        VACMS-11761-Delay-accessibility-tests-to-allow-CSS-transitions-to-finish PASS
                                     VACMS-11811-Shifts-PHP-Lint-into-GHA-action PASS
                             VACMS-11881-Change-order-of-file-permission-changes PASS
              VACMS-11896-Make-contentreleasestatuscontroller-test-more-reliable PASS
                                            VACMS-11904-Undefined-array-key-href PASS
                                          VACMS-11905-Undefined-key-req_optional PASS
                            VACMS-11906-Patch-taxonomy_entity_index-to-fix-error PASS
                                                               VACMS-8671-Part-2 PASS
                 dependabot/composer/drupal/dynamic_entity_reference-3.0.0-beta1 PASS
                                     dependabot/composer/aws/aws-sdk-php-3.250.0 PASS
                                     dependabot/composer/aws/aws-sdk-php-3.252.5 PASS
                                     dependabot/composer/aws/aws-sdk-php-3.253.1 PASS
                                     dependabot/composer/aws/aws-sdk-php-3.254.0 PASS
                            dependabot/composer/consolidation/site-process-4.2.1 PASS
                             dependabot/composer/cweagans/composer-patches-1.7.2 PASS
                                     dependabot/composer/datadog/dd-trace-0.79.0 PASS
                                     dependabot/composer/datadog/dd-trace-0.80.0 PASS
                                dependabot/composer/drupal/allowed_formats-1.5.0 PASS
                                   dependabot/composer/drupal/animated_gif-1.7.0 PASS
                         dependabot/composer/drupal/auto_entitylabel-3.0.0-beta4 PASS
                               dependabot/composer/drupal/config_split-2.0.0-rc4 PASS
                              dependabot/composer/drupal/core-recommended-9.3.11 PASS
                              dependabot/composer/drupal/core-recommended-9.3.12 PASS
                              dependabot/composer/drupal/core-recommended-9.3.13 PASS
                              dependabot/composer/drupal/core-recommended-9.3.14 PASS
                              dependabot/composer/drupal/core-recommended-9.3.15 PASS
                               dependabot/composer/drupal/core-recommended-9.3.9 PASS
                               dependabot/composer/drupal/core-recommended-9.4.1 PASS
                               dependabot/composer/drupal/core-recommended-9.4.3 PASS
                                   dependabot/composer/drupal/ctools_block-4.0.1 PASS
                                          dependabot/composer/drupal/danse-2.2.7 PASS
                 dependabot/composer/drupal/danse_content_moderation-2.0.0-beta3 PASS
                                          dependabot/composer/drupal/devel-4.1.5 PASS
                                     dependabot/composer/drupal/dropzonejs-2.7.0 PASS
                                   dependabot/composer/drupal/entity_embed-1.1.0 PASS
                                dependabot/composer/drupal/fast_404-2.0.0-alpha5 PASS
                                      dependabot/composer/drupal/geocoder-3.31.0 PASS
                          dependabot/composer/drupal/hook_event_dispatcher-3.3.1 PASS
                               dependabot/composer/drupal/menu_breadcrumb-1.13.0 PASS
                                       dependabot/composer/drupal/metatag-1.22.0 PASS
                             dependabot/composer/drupal/migrate_source_csv-3.5.0 PASS
                                           dependabot/composer/drupal/next-1.3.1 PASS
                                           dependabot/composer/drupal/next-1.4.0 PASS
                                           dependabot/composer/drupal/next-1.6.0 PASS
                              dependabot/composer/drupal/node_link_report-1.18.0 PASS
                                dependabot/composer/drupal/password_policy-3.1.0 PASS
                           dependabot/composer/drupal/path_redirect_import-2.0.5 PASS
                                      dependabot/composer/drupal/pathauto-1.10.0 PASS
                                      dependabot/composer/drupal/pathauto-1.11.0 PASS
                                       dependabot/composer/drupal/pathauto-1.9.0 PASS
                                       dependabot/composer/drupal/redirect-1.6.0 PASS
                                dependabot/composer/drupal/role_delegation-1.1.0 PASS
                                                    el-delete-composer-cache-dir FAIL
                                             el-php-81-disable-memcache-brd-docs FAIL
                               integration-VACMS-8877-homepage-hardening-refresh FAIL
                                                        integration-covid-status FAIL
                                                              integration-lovell FAIL
                                                    integration-recurring-events FAIL
                                                             integration-tour-ui FAIL
                                                   joshworkscivicactions-patch-1 FAIL
                                                                            main FAIL
                                                                          master FAIL
                                                             mchelen-gov-patch-1 FAIL
                                                                       neil-demo FAIL
                                                                 next-drupal-poc FAIL
                                                             nh_composer_package FAIL
                        revert-10362-dependabot/composer/drupal/smart_date-3.6.0 PASS
                                revert-11949-VACMS-11948-gha_command_deprecation PASS
                                          revert-9089-VACMS-8649-next_news_story PASS
                           revert-9369-revert-9345-VACMS-9097-new-facility-email PASS
                                 revert-9459-revert-9401-VACMS-8588-remove-lando PASS
                                                                   VACMS-invalid FAIL
                                                   VAGOV_TEAM-2304204923-invalid FAIL
```

### CODE REVIEW

**Review by timcosgrove — APPROVED** [review]
THis looks reasonable
